### PR TITLE
feature expose skip state refresh on plan phase to our CLI

### PR DESCRIPTION
**Pull request by sabasabo**
### Issue & Steps to Reproduce / Feature Request
Adding skipStateRefresh arg to destroy

#### Expected Result / DoD ✅:
So we can also use this feature in our stage per PR process we need to expose this feature to our CLI

`env0 destroy ... -z true ...` will use the skipStateRefresh param

### CODE REVIEW

**Comment by sabasabo** [comment]
[feature expose skip state refresh on plan phase to our CLI](https://trello.com/c/2iFnp6lv/5506-feature-expose-skip-state-refresh-on-plan-phase-to-our-cli)

File: node/README.md [hunk 1]
```diff
@@ -143,6 +148,7 @@ https://docs.env0.com/reference
 You can compile and run the CLI from source by cloning the repo from Github and then running the following:
 ```bash
 # clone the repo from github
+# make sure env0 lib is not installed globally
 yarn install
 yarn link # link your local copy of the CLI to your terminal path
```
Comment (sagilaufer1992) [inline node/README.md:153]: doesn't it bypasses the globally installed comment? or the terminal first looks at global commands?

File: node/src/commands/deploy.js [hunk 2]
```diff
@@ -36,8 +36,8 @@ const deploy = async (options, variables) => {
     deployment = environment.latestDeploymentLog;
   } else {
     assertNoWorkspaceNameChanges(options, environment);
-    const optionsWithoutWorkspace = { ...options }
-    delete optionsWithoutWorkspace[WORKSPACE_NAME]
+    const optionsWithoutWorkspace = { ...options };
+    delete optionsWithoutWorkspace[WORKSPACE_NAME];
```
Comment (sagilaufer1992) [inline node/src/commands/deploy.js:40]: there's no prettier/linter I presume? if so , should we open a ticket for next PRs?

File: node/tests/commands/destroy.spec.js [hunk 3]
```diff
@@ -29,17 +29,18 @@ describe('destroy', () => {
...
 
     await destroy({});
 
-    expect(mockDestroyEnvironment).toBeCalledWith(mockEnvironment);
+    expect(mockDestroyEnvironment).toHaveBeenCalledWith(mockEnvironment, options);
     expect(mockPollDeploymentStatus).toBeCalledWith(mockDeployment);
   });
 
-  it('should fail when environment doesnt exist', async () => {
+  it("should fail when environment does'nt exist", async () => {
```
Comment (sagilaufer1992) [inline node/tests/commands/destroy.spec.js:43]:
```suggestion
  it("should fail when environment doesn't exist", async () => {
```

File: node/tests/lib/deploy-utils.spec.js [hunk 4]
```diff
@@ -45,6 +53,19 @@ describe('deploy utils', () => {
     });
   });
 
+  describe('skip automatic state refresh', () => {
+    it.each`
+      params                              | sentParams
+      ${{ [SKIP_STATE_REFRESH]: 'true' }} | ${{ params: { [SKIP_STATE_REFRESH]: 'true' } }}
+      ${{}}                               | ${{ params: {} }}
```
Comment (sagilaufer1992) [inline node/tests/lib/deploy-utils.spec.js:60]: could use `{ params }`, no need for another sentParams variable that is used once in that manner

File: node/tests/commands/destroy.spec.js [hunk 5]
```diff
@@ -49,6 +50,26 @@ describe('destroy', () => {
     );
   });
 
+  describe('skipStateRefresh argument', () => {
+    it.each`
+      skipStateRefreshOption | existing
+      ${'true'}              | ${'true'}
+      ${'false'}             | ${'false'}
+      ${undefined}           | ${undefined}
```
Comment (sagilaufer1992) [inline node/tests/commands/destroy.spec.js:58]: if they are the same, why not creating a single `options` and expect the function to be called with it (like in proper data test above)? why distinguish the passed parameter and the expected?

File: node/src/lib/deploy-utils.js [hunk 6]
```diff
@@ -76,10 +76,12 @@ class DeployUtils {
     });
   }
 
-  async destroyEnvironment(environment) {
+  async destroyEnvironment(environment, params) {
     logger.info('Starting to destroy environment...');
 
-    return await apiClient.callApi('post', `environments/${environment.id}/destroy`);
+    return await apiClient.callApi('post', `environments/${environment.id}/destroy`, {
+      params: params
```
Comment (sagilaufer1992) [inline node/src/lib/deploy-utils.js:83]:
```suggestion
      params
```

File: node/src/commands/destroy.js [hunk 7]
```diff
@@ -25,7 +25,10 @@ const destroy = async options => {
     await deployUtils.updateEnvironment(environment, { requiresApproval: requiresApproval });
   }
 
-  const deployment = await deployUtils.destroyEnvironment(environment);
+  const params = removeEmptyValuesFromObj({
+    [SKIP_STATE_REFRESH]: options[SKIP_STATE_REFRESH]
+  });
```
Comment (sagilaufer1992) [inline node/src/commands/destroy.js:30]: why bother? what's the harm in undefined here? it seems the api wii manage, so it is kind of redundant, no? no need for `removeEmprtyValuesFromObj` IMO

**Review by sagilaufer1992 — COMMENTED** [review]
LGTM, minor comments.

File: node/README.md (hunk 1 above)
Comment (sabasabo) [reply node/README.md:153]: The terminal gives the global command an higher priority 😭

File: node/src/commands/deploy.js (hunk 2 above)
Comment (sabasabo) [reply node/src/commands/deploy.js:40]: Yes

File: node/src/config/arguments.js [hunk 8]
```diff
@@ -119,6 +120,13 @@ const argumentsMap = {
     group: ['deploy'],
     description: 'Your git revision, can be a branch tag or a commit hash. Default value "master" '
   },
+  [SKIP_STATE_REFRESH]: {
+    name: SKIP_STATE_REFRESH,
+    alias: 'z',
+    type: String,
```
Comment (eranelbaz) [inline node/src/config/arguments.js:126]: no `	type: bool`?

**Review by eranelbaz — CHANGES_REQUESTED** [review]
Agree with @sagilaufer1992 's comments
but currently requesting changes since you didn't bump package version

File: node/src/commands/deploy.js (hunk 2 above)
Comment (sabasabo) [reply node/src/commands/deploy.js:40]: Created: 
[Add Husky and Github lint checker to the CLI repo on env0 - RnD | Trello](https://trello.com/c/ZyEK7h8q/5718-add-husky-and-github-lint-checker-to-the-cli-repo) https://trello.com/c/ZyEK7h8q/5718-add-husky-and-github-lint-checker-to-the-cli-repo

File: node/src/commands/destroy.js (hunk 7 above)
Comment (sabasabo) [reply node/src/commands/destroy.js:30]: In normal usage, you don't want to see prints from the cli such as: sending `param: undefined` to api-route

Unless the user explicitly used the param, it should not be sent.

File: node/src/config/arguments.js (hunk 8 above)
Comment (sabasabo) [reply node/src/config/arguments.js:126]: Type bool is checking if the flag exists or not. `env0 -z` is `true` and `env0` is `false`
Using the string we can be more explicit like in `requiresApproval`

File: node/tests/lib/deploy-utils.spec.js (hunk 4 above)
Comment (sabasabo) [reply node/tests/lib/deploy-utils.spec.js]: Done: 36cdd6b6a2cb7d845c29dd480ff7dfa9516b16fd

File: node/tests/commands/destroy.spec.js (hunk 5 above)
Comment (sabasabo) [reply node/tests/commands/destroy.spec.js]: Done: 36cdd6b6a2cb7d845c29dd480ff7dfa9516b16fd

File: node/src/lib/deploy-utils.js (hunk 6 above)
Comment (sabasabo) [reply node/src/lib/deploy-utils.js]: Done: 36cdd6b6a2cb7d845c29dd480ff7dfa9516b16fd

**Comment by sagilaufer1992** [comment]
LGTM. bump a version according to @eranelbaz request, I'll approve

**Comment by sabasabo** [comment]
> LGTM. bump a version according to @eranelbaz request, I'll approve

Missed that 😅
Bumped the version: 77a188966249196217e9ae86bd819a04f10b5795

**Review by sagilaufer1992 — APPROVED** [review]

**Pull request merged**